spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/SupportedBindableFeatures.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.function;
+
+/**
+ * Internal abstraction for the supported bindable features.
+ *
+ * @author Soby Chacko
+ * @since 4.0.0
+ */
+class SupportedBindableFeatures {
+
+	private final boolean pollable;
+
+	private final boolean reactive;
+
+	SupportedBindableFeatures(boolean pollable, boolean reactive) {
```

Comment:
I am now wondering if we should just have setters/getters for features. I know at the moment we're calling out these two as features, but as discussed before there will be more, so we don't really want keep on changing this constructor